src/main/java/org/peergos/client/NabuClient.java

```diff
@@ -0,0 +1,265 @@
+        this.host = host;
+        this.port = port;
+        this.connectTimeoutMillis = connectTimeoutMillis;
+        this.readTimeoutMillis = readTimeoutMillis;
+
+        if (ssl) {
+            this.protocol = "https";
+        } else {
+            this.protocol = "http";
+        }
+
+        this.apiVersion = version;
+    }
+
+    public NabuClient timeout(int timeout) {
+        return new NabuClient(host, port, apiVersion, timeout, timeout, protocol.equals("https"));
+    }
+
+    public String version() throws IOException {
+        Map m = (Map)retrieveAndParse("version");
+        return (String)m.get("Version");
+    }
+
+    public Map id() throws IOException {
+        return retrieveMap("id");
+    }
+
+    public List<Multihash> listBlockstore() throws IOException {
+        String jsonStream = new String(retrieve("refs/local"));
+        return JSONParser.parseStream(jsonStream).stream()
+                .map(m -> (String) (((Map) m).get("Ref")))
+                .map(Cid::decode)
+                .collect(Collectors.toList());
+    }
+
+    public boolean hasBlock(Multihash hash, Optional<String> auth) throws IOException {
+        String authArg = auth.isPresent() ? "&auth=" + auth.get() : "";
+        return "true".equals(new String(retrieve("block/has?arg=" + hash + authArg)));
+    }
+
+    public boolean bloomAdd(Multihash hash) throws IOException {
+        return "true".equals(new String(retrieve("bloom/add?arg=" + hash)));
+    }
+
+    public byte[] getBlock(Multihash hash, Optional<String> auth) throws IOException {
+        String authArg = auth.isPresent() ? "&auth=" + auth.get() : "";
+        return retrieve("block/get?arg=" + hash + authArg);
+    }
+
+    public void removeBlock(Multihash hash) throws IOException {
+        retrieve("block/rm?arg=" + hash);
+    }
+
+    public List<Cid> putBlock(List<byte[]> data) throws IOException {
+        return putBlocks(data, Optional.empty());
+    }
+
+    public List<Cid> putBlocks(List<byte[]> data, Optional<String> format) throws IOException {
+        List<Cid> res = new ArrayList<>();
+        for (byte[] value : data) {
+            res.add(putBlock(value, format));
+        }
+        return res;
+    }
+
+    public Cid putBlock(byte[] data, Optional<String> format) throws IOException {
+        String fmt = format.map(f -> "&format=" + f).orElse("");
+        Multipart m = new Multipart(protocol +"://" + host + ":" + port + apiVersion+"block/put?stream-channels=true" + fmt, "UTF-8");
+        try {
+            m.addFilePart("file", Paths.get(""), new NamedStreamable.ByteArrayWrapper(data));
+            String res = m.finish();
+            LinkedHashMap<String, String> obj = (LinkedHashMap<String, String>)JSONParser.parse(res);
+            return Cid.decode(obj.get("Hash"));
+        } catch (IOException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
+    public Map stat(Multihash hash) throws IOException {
```

Comment:
This basically just returns the size right?

Reply:
Yes, now just returning the size as an int